### ISSUE TITLE
mldonkey failed to build on 10.10 [fixed]

### ISSUE THREAD

**Issue by korsmakolnikov** [Homebrew/homebrew#41256]
brew install mldonkey
==> Downloading https://downloads.sourceforge.net/project/mldonkey/mldonkey/3.1.5/mldonkey-3.1.5.tar.bz2
Already downloaded: /Library/Caches/Homebrew/mldonkey-3.1.5.tar.bz2
==> ./configure --prefix=/usr/local/Cellar/mldonkey/3.1.5
********  Objective-Caml 3.12.0 is required  *********
*******  Check http://caml.inria.fr/  ********
Do you want this script to try to download and install ocaml
LOCALLY in mldonkey directory ?
./configure: line 4943: read: read error: 0: Bad file descriptor

**Comment by jacknagel** [comment]
https://git.io/brew-troubleshooting

**Comment by mistydemeo** [comment]
It looks like mldonkey needs camlp4 to build. I can repro.

If I add that dependency, however, it fails to build like so: https://gist.github.com/674a243b206a26e57e9f

```
File "src/utils/net/terminal.ml", line 188, characters 2-4:
Error: This comment contains an unterminated string literal
File "src/utils/net/terminal.ml", line 188, characters 5-7:
Error: String literal begins here
```

There's an upstream bug report here, looks like it's been patched: https://savannah.nongnu.org/bugs/?43934

**Comment by mistydemeo** [comment]
Fix in #41261!

**Comment by korsmakolnikov** [comment]
File "src/utils/extlib/IO.ml", line 747, characters 4-21:
Warning 3: deprecated: String.unsafe_set
File "src/utils/extlib/IO.ml", line 1:
Error: Could not find the .cmi file for interface src/utils/extlib/IO.mli.
make: *** [src/utils/extlib/IO.cmx] Error 2
